### ISSUE TITLE
retireOneExport should not delete valToSlot

### ISSUE THREAD

**Issue by warner** [Agoric/Agoric#3978]
As @FUDCo and I were reviewing #3732 , in particular https://github.com/Agoric/agoric-sdk/pull/3968#discussion_r729187128 , we concluded that the `retireOneExport()` function in liveslots is currently doing the wrong thing. When the kernel retires a Remotable, liveslots should simple remove that vref from `kernelRecognizableRemotables`, and *not* touch `valToSlot` or `slotToVal`.

Previously (before virtual objects), I was using the presence or absence of a vref in `slotToVal` as an indicator of the kernel's interest in an object. Back then, if the kernel couldn't recognize it, we didn't need to maintain the vref-to-Remotable mapping: re-exporting the object could give it a new vref and the kernel couldn't tell the difference (because it had already forgotten the old vref, when it did the `retire`).

But now, the following sequence might happen:
* a Remotable is created, and exported to the kernel
  * serializing the new Remotable causes a vref to be assigned and `valToSlot`/`slotToVal` to be populated
* the same Remotable gets stored in some virtualized data
  * this adds the vref to the DB, and increments a refcount for it
* the kernel drops and retires the exported vref
* someone stores the Remotable in some other bit of virtualized data (serializing it), and the serialized data should use the same vref as before
* or, someone reads the old virtualized data, deserializes the old vref, and expects to get back the same Remotable as before

If `retireOneExport` were to delete the `valToSlot` entry, that last `serialize` would allocate a new vref, and that last `deserialize` would crash because the stored vref was no longer in `slotToVal`.

The fix will be to replace that entire "if (isRemotable)" clause of `retireOneExport` with a simple `kernelRecognizableRemotables.delete(vref)`, and leave slotToVal/valToSlot alone. And the only purpose of `kernelRecognizableRemoteables` is to permit/inhibit the `syscall.retireExports` during `processDeadSet`.

The more interesting part will be to write the unit test for this. It ought to follow the sequence above: first attempting to deserialize the old VO (and asserting that it doesn't crash), then serialize the old Remotable into a new VO and compare the backing store / DB entries to assert that the vrefs are the same.

**Comment by warner** [comment]
We ran into this again in #4679, and spent an afternoon figuring it out (again).

**Comment by FUDCo** [comment]
After creating a test that manifests the bug, I've verified that removing the `slotToVal`/`valToSlot` deletions from `retireOneExport` makes the bug go away.  However, I'm fairly sure that the prescription given in this issue is wrong, and that we really do need something like the more complicated recipe we cooked up in #4679.  I think the proposed simplification of `retireOneExport` introduces a storage leak; the current code is the only thing that removes `slotToVal` entries when exports are retired, so without it these entries will accumulate in memory.

**Comment by warner** [comment]
@FUDCo walked through this and we decided that the correct process is indeed "for Remotables, do nothing except remove from `kernelRecognizableRemotables`". The Remotable is tracked in the FinalizationRegistry along with Presences and Representatives, and will be cleaned up from both `slotToVal` and `valToSlot` when it goes away (`valToSlot` is a WeakMap, so that's automatic, and `slotToVal` is handled in the FR's callback when it sees the WeakRef is empty).

However, that FinalizationRegistry is currently named `droppedImportRegistry`, which is confusing, because it's used for more than just imports. That registry tracks Presences (imported), Remotables (exported and/or referenced by virtual data), and Presences (same). We decided to rename it to `vreffedObjectRegistry`, which is kinda goofy but clear: this is a FinalizationRegistry which tracks all (non-Promise) JS `Object`s which are associated with vrefs.
